BennyTordrupVisma/Week2/Ass3/script.js

```diff
@@ -0,0 +1,340 @@
+	},
+	
+	archiveMessage: function(message) {
+		this.props.onArchiveMessage(message, this.props.list);
+	},
+	
+	unarchiveMessage: function(message) {
+		this.props.onUnarchiveMessage(message, this.props.list);
+	},
+})
+
+var InputField = React.createClass({
+	render: function() {
+		return <div>
+				<label>Text to add: </label>
+				<input type="text" ref="input"/>
+				<label>Name of list: </label>
+				<input type="text" ref="listInput"/>
+				<button onClick={this.handleCommit}>Commit</button>
+			</div>;
+	},
+	
+	isInputValid: function(input, list) {
+		if (input.length > 200)
+		{
+			window.alert("The input may not exceed 200 characters.");
+			return false;
+		}
+		
+		if (list.length == 0)
+		{
+			window.alert("You have to enter a name on the list to add message to.");
+			return false;
+		}
+		
+		if (this.props.lists.some(l => l.messages.some(m => m.text == input)))
+		{
+			window.alert("The message is already member of a list and cannot be added");
+			return false;
+		}
+		
+		return true;
+	},
+	
+	handleCommit: function() {
+		var input = this.refs.input.value;
+		var list = this.refs.listInput.value;
+		if (this.isInputValid(input, list)) {
+			this.props.handleCommit(input, list);
+			this.refs.input.value = '';
+			this.refs.listInput.value = '';
+			this.refs.input.focus();
+		}
+	},
+});
+
+var App = React.createClass({
+	getInitialState: function() {
+		return {
+			//lists: []
+			lists: [{
+				name: "List 1", 
+				messages: [
+					{
+						id: 1,
+						text: "Test 1-1"
+					},
+					{
+						id: 2,
+						text: "Test 1-2",
+						isArchived: true
+					},
+					{
+						id: 3,
+						text: "Test 1-3"
+					},
+					{
+						id: 4,
+						text: "Test 1-4",
+						isArchived: true
+					},
+					{
+						id: 5,
+						text: "Test 1-5"
+					}] 
+			},
+			{
+				name: "List 2", 
+				messages: [
+					{
+						id: 1,
+						text: "Test 2-1"
+					},
+					{
+						id: 2,
+						text: "Test 2-2",
+						isArchived: true
+					},
+					{
+						id: 3,
+						text: "Test 2-3"
+					},
+					{
+						id: 4,
+						text: "Test 2-4",
+						isArchived: true
+					},
+					{
+						id: 5,
+						text: "Test 2-5"
+					}] 
+			}]
+		}
+	},
+	
+	render: function() {
+		var that = this;
+		var listList = this.state.lists.map(function(list, index) {
+			return <List key={index} 
+						list={list} 
+						onArchiveMessage={that.archiveMessage} 
+						onDeleteMessage={that.deleteMessage}
+						onUnarchiveMessage={that.unarchiveMessage}
+						onMoveMessage={that.moveMessage} />
+		});
+		
+		return 	<div>
+					<InputField handleCommit={this.commitMessage} lists={this.state.lists}/>
+					<div>
+						<h3>Lists</h3>
+						<div>
+							{listList}
+						</div>
+					</div>
+				</div>
+	},
+
+	moveMessage: function(message, oldList, newListName) {
+		console.log("Moving message from " + oldList.name + " to " + newListName)
+		var sourceList = this.state.lists.find(l => l.name == oldList.name);
+		if (sourceList == null)
```

Comment:
You can drop `== null`; `if (sourceList)` will do just fine as long as `sourceList` has been declared (which in this case, it has). An undefined variable in JavaScript, when converted to boolean will evaluate to `false`.

A small sidenote: If the desired list is not found in the array, the `sourceList` will be `undefined`.